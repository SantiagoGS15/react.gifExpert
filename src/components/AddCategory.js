import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
  
    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue?.trim().length > 2){
            setCategories(categories => [inputValue, ...categories]);
            setInputValue('');
        }
    }

    return (
    <form onSubmit={handleSubmit}>
        <h1>Add Category</h1>
        <input 
            value={inputValue}
            onChange={handleInputChange}

            type= "text"
        />
    </form>);
};


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};