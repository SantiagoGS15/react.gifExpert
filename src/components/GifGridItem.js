import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {
  return (<div className="card">
      <img src={url} alt={title}></img>
      <p className="animate__animated animate__bounce">{title}</p>
  </div>);
};


GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};