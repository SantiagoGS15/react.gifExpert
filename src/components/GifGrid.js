import React from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';

import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category}) => {

    const {data: images, loading} = useFetchGifs(category);
  
    return (
      <>
        <h3 className="animate__fadeIn">{category}</h3>
        {loading && <p>Loading...</p>}
        <div className="card-grid">
          
          <ol>
            {
              images.map((item) => {
                return <GifGridItem 
                      {...item}
                      key={item.id}>
                    </GifGridItem>
              })
            }
          </ol>
        </div>
      </>
  );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};
