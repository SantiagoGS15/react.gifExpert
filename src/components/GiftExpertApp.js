import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';
import PropTypes from 'prop-types';

export const GiftExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories);
    
    return (<>
        <h2>GiftExpertApp</h2>
        <AddCategory setCategories={setCategories}></AddCategory>
        <hr></hr>
        
        <ol>
            {
               categories.map((category) => {
                   return <GifGrid category={category} key={category}/>
               })
            }
        </ol>
    </>);
};

GiftExpertApp.propTypes = {
    defaultCategories: PropTypes.array
}