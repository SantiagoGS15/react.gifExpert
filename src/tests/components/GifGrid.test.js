import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <AddCategory>', () => {
    const category = 'One Punch';
    

    test('Debe mostrarse correctamente', () => {
        useFetchGifs.mockReturnValue(
            {
                data: [],
                loading: true
            }
        );
        const wrapper = shallow(<GifGrid category={category}></GifGrid>);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
        const mockDataImages = [
            {
                id: 1,
                url: 'http://localhost/q.png',
                title: 'cualquier cosa'
            }
        ]
        useFetchGifs.mockReturnValue(
            {
                data: mockDataImages,
                loading: false
            }
        );
        const wrapper = shallow(<GifGrid category={category}></GifGrid>);
        expect(wrapper).toMatchSnapshot();

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(mockDataImages.length);
    });

    
});