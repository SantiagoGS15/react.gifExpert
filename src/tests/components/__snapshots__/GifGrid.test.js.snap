// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__fadeIn"
  >
    One Punch
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  >
    <ol />
  </div>
</Fragment>
`;

exports[`Pruebas en <AddCategory> debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__fadeIn"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <ol>
      <GifGridItem
        id={1}
        key="1"
        title="cualquier cosa"
        url="http://localhost/q.png"
      />
    </ol>
  </div>
</Fragment>
`;
